python/pyiceberg/transforms.py

```diff
@@ -644,8 +751,89 @@ def can_transform(self, _: IcebergType) -> bool:
     def result_type(self, source: IcebergType) -> IcebergType:
         return source
 
+    def project(self, name: str, pred: BoundPredicate) -> Optional[UnboundPredicate]:
+        return None
+
     def to_human_string(self, _: IcebergType, value: Optional[S]) -> str:
         return "null"
 
     def __repr__(self) -> str:
         return "VoidTransform()"
+
+
+M = TypeVar("M")
+
+
+def _truncate_number(
+    name: str, pred: BoundLiteralPredicate, transform: Callable[[Optional[M]], Optional[M]]
```

Comment:
Should this be `BoundLiteralPredicate[M]`?

Reply:
That's correct! We won't be able to forget this after https://github.com/apache/iceberg/pull/6232

Reply:
We can also re-use `L` here and drop `M`